src/app/catalogue/Catalogue.java

```diff
@@ -77,18 +79,19 @@ private static void printSearchResults(int count, String output) {
     public static List<Item> searchByTitle(List<Item> list, String titleSearch) {
         List<Item> newList = new ArrayList<>();
 
-        if (list != null && !list.isEmpty()) {
-            for (Item item : list) {
-                if (item != null && item.getTitle().toLowerCase().contains(titleSearch.toLowerCase())
-                        || !"".equals(titleSearch)) {
-                    newList.add(item);
-                } else {
-                    return newList;
-                }
-            }
+        if (list == null || list.isEmpty()) {
             return newList;
         }
-        return null;
+
+        for (Item item : list) {
+            if (item != null && item.getTitle().toLowerCase().contains(titleSearch.toLowerCase())
```

Comment:
But here we are returning a new list of items that contain whatever is in the search string from the original list of items.